apidef/oas/default.go

```diff
@@ -8,19 +8,34 @@ import (
 )
 
 const (
-	invalidServerURLFmt = "Please update %q to be a valid url or pass a valid url with upstreamURL query param"
+	invalidServerURLFmt       = "Please update %q to be a valid url or pass a valid url with upstreamURL query param"
+	MiddlewareValidateRequest = "validateRequest"
+	MiddlewareAllowList       = "allowList"
 )
 
 var (
 	errEmptyServersObject = errors.New("servers object is empty in OAS")
 	errInvalidUpstreamURL = errors.New("invalid upstream URL")
 	errInvalidServerURL   = errors.New("Error validating servers entry in OAS")
+	allowedMethods        = []string{
+		http.MethodConnect,
+		http.MethodDelete,
+		http.MethodGet,
+		http.MethodHead,
+		http.MethodOptions,
+		http.MethodPatch,
+		http.MethodPost,
+		http.MethodPut,
+		http.MethodTrace,
+	}
 )
 
 type TykExtensionConfigParams struct {
-	UpstreamURL  string
-	ListenPath   string
-	CustomDomain string
+	UpstreamURL     string
+	ListenPath      string
+	CustomDomain    string
+	AllowList       *bool
+	ValidateRequest *bool
```

Comment:
can this be just type `bool`, I'm missing why it's a pointer

Reply:
params `validateRequest`, `allowList`, `authentication` has both `true` and `false` as valid values, we have to distinguish when the parameter is not provided, hence the pointer